generate/templates/main.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"archive/zip"
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+func main() {
+	var zipBuff bytes.Buffer
+	zipWriter := zip.NewWriter(&zipBuff)
+	err := filepath.Walk("templates", func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if filepath.Clean(path) == "templates" {
+			return nil
+		}
+
+		name, err := filepath.Rel("templates", path)
+		if err != nil {
+			return err
+		}
+		name = filepath.ToSlash(name)
+		if info.IsDir() {
+			name = name + "/"
+		}
+
+		header := &zip.FileHeader{
+			Name:   name,
+			Method: zip.Deflate,
+		}
+		header.SetMode(info.Mode())
+
+		w, err := zipWriter.CreateHeader(header)
+		if err != nil {
+			return err
+		}
+
+		if !info.IsDir() {
+			b, err := os.ReadFile(path)
+			if err != nil {
+				return err
+			}
+
+			_, err = w.Write(b)
+			if err != nil {
+				return err
+			}
+		}
+
+		return nil
+	})
+	zipWriter.Close()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	f, err := os.OpenFile("zz_filesystem_generated.go", os.O_TRUNC|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+
+	srcOut := bufio.NewWriter(f)
+	defer srcOut.Flush()
+
+	_, err = fmt.Fprintf(srcOut, "package function\n\nvar templatesZip = []byte{\n")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	zipBytesReader := bytes.NewReader(zipBuff.Bytes())
+	buff := make([]byte, 32)
+	for {
+		n, err := zipBytesReader.Read(buff)
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				break
+			} else {
+				log.Fatal(err)
+			}
+		}
+
+		_, err = fmt.Fprintf(srcOut, "\t")
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		for i, b := range buff[:n] {
+			_, err = fmt.Fprintf(srcOut, "0x%02x,", b)
```

Comment:
This is actually what prompted me to ask for some comments. Do I understand correctly that this writes each byte to the `srcOut` writer in hex, with a `0x` prefix? It took me a minute to mentally parse all of the x's and 0's.

Reply:
yes, %02x -- means hex format with width padding